### DISCUSSION TITLE
HelmChart refresh after update of included git repository

### DISCUSSION THREAD

**Discussion by Xeinn** [fluxcd/flux2#2630] [Q&A] [answered]
I've been experimenting with Fluxv2 and have come across a situation that I'm not sure if it is a bug or expected behaviour

I have a git repository containing a values file & a second git repository defining a helm chart.

I have created a flux GitRepository source (A) which references the helm chart
I have a second GitRepository source (B) that references the git repo containing the values file and additionally includes the helm chart GitRepository source

I then have a HelmRelease that references the GitRepository source (B) (which should contain the values file and helm chart) - HelmRelease is set to reconcileStrategy "Revision"

When I push a change to the repository containing the helm chart, the including git repository is refreshed (as per the documentation) but the HelmRelease that references the GitRepository source does not trigger the refresh of the HelmChart with new details (I suspect that this is due to the HelmRelease not seeing a change in the GitRepository release identifier which is constructed from helm chart version and the git repo checksum - which doesn't change since the including git repo hasn't actually changed even if the included repo has)

Appreciate this may be a really fringe case - as I say I'm just experimenting with configurations - but I would have expected the HelmChart to be updated as a result of the changes in the included repo.

The reason for the odd include approach is so users with access to the repo containing the value file can't use it to deploy anything else - they can only provide values for the chart repo controlled by another team.

**Comment by kingdonb** [comment, marked as answer]
https://fluxcd.io/docs/guides/helmreleases/#refer-to-values-in-configmaps-generated-with-kustomize

You may need to see this... ☝️ 

I would think that we should test the specific edge case that you think you have, but can you please provide your example in code form so we are sure we are testing the same thing? I would expect the right thing to happen and for GitRepository sources for HelmReleases to always get a new "SHA" for their downstream revisions' use when `include`d repositories are updated.

(Does it catch up eventually if you reconcile the child repository, then the parent gitrepository, then reconcile the downstream HelmRelease? Or does it stay in this "behind" state forever?)

If you can adapt your approach to the example I've provided, that may entirely solve your issue. If there's something I did not understand, please let me know and I'll take a second look.

Another possibility: reverse the order, make the one that doesn't change much the included one. You should have the freedom to change the parent or the child, just trying to suggest a workaround that might get you un-blocked in case there is a real bug here.
